agent/lib/artemis.js

```diff
@@ -523,6 +523,22 @@ Artemis.prototype.removeAddressSettings = function (match) {
     return this._request('broker', 'removeAddressSettings', [match]);
 };
 
+Artemis.prototype.getAddressSettings = function (match) {
+    return this._request('broker', 'getAddressSettingsAsJSON', [match]).then(function (result) {
+        return JSON.parse(result);
+    });
+};
+
+Artemis.prototype.listAddressSettings = function () {
+    var self = this;
+    return this.listAddresses().then(function (addresses) {
+        return Promise.all(addresses.map(function (a) {
```

Comment:
suggest using the `plimit` pattern to reduce the inflight.

Reply:
I'll remove that method.  broker_controller already has the list of addresses, and is using that list to call getAddressSettings (with p-limit).  
This one was the original, that is no longer used.

Reply:
ok